sqlmesh/core/audit/definition.py

```diff
@@ -352,7 +354,7 @@ def metadata_hash(self, audits: t.Dict[str, ModelAudit]) -> str:
         ]
 
         query = self.render_query(self) or self.query
-        data.append(query.sql(comments=False))
+        data.append(query.sql(dialect=self.dialect, comments=False))
```

Comment:
this is a breaking change if you do this so need to switch to feat!

maybe should switch this to gen?